app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_post_counter
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
- [ Optional ] Kindly I recommend you  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private Just like this :- 

  # app/models/post.rb
```
  after_save :update_post_counter

  private

  def update_post_counter
        user.update(posts_counter: user.posts.count)
  end
```